### PR TITLE
Editing issue with pickle def with lambda function

### LINKED ISSUE #23865: Possible pickle issues
https://github.com/huggingface/transformers/blob/af2aac51fc1c59237ff7228908ace2cd8fc0d9a6/src/transformers/optimization.py#L49
When attempting to pickle this function, there is a potential for an error to occur due to the lambda function that cannot be pickled.
I suggest the following solution for this problem.

```
def get_constant_lambda(_):
    return 1

def get_constant_schedule(optimizer: Optimizer, last_epoch: int = -1):
    """
    Create a schedule with a constant learning rate, using the learning rate set in optimizer.

    Args:
        optimizer ([`~torch.optim.Optimizer`]):
            The optimizer for which to schedule the learning rate.
        last_epoch (`int`, *optional*, defaults to -1):
            The index of the last epoch when resuming training.

    Return:
        `torch.optim.lr_scheduler.LambdaLR` with the appropriate schedule.
    """

    return LambdaLR(optimizer, get_constant_lambda, last_epoch=last_epoch)
```

### PR DESCRIPTION
**Pull request by Natyren**
# What does this PR do?
In this PR, I address the problem of pickling the constant LR scheduler, which fails during the process (potentially during multi-GPU training, as observed in my case) due to the presence of a lambda function within it.

Fixes  #23865    (issue)

### CODE REVIEW

File: src/transformers/optimization.py [hunk 1]
```diff
@@ -32,6 +32,21 @@
 logger = logging.get_logger(__name__)
 
 
+def get_constant_lambda(_=None):
```
Comment (sgugger) [inline src/transformers/optimization.py:35]:
```suggestion
def _get_constant_lambda(current_step):
```

File: src/transformers/optimization.py [hunk 2]
```diff
@@ -32,6 +32,21 @@
...
+    """
+    Return 1, independent from args.
+
+    Args:
+        _ ( *optional*, defaults to None):
+            Placeholder to argument, used to consistency with args in LambdaLR
+    
+    Return:
+        1 : int - constant lambda for constant scheduler
+    """
```
Comment (sgugger) [inline src/transformers/optimization.py:45]: No need for a documentation for a private function.

File: src/transformers/optimization.py [hunk 3]
```diff
@@ -46,7 +61,7 @@ def get_constant_schedule(optimizer: Optimizer, last_epoch: int = -1):
         `torch.optim.lr_scheduler.LambdaLR` with the appropriate schedule.
     """
 
-    return LambdaLR(optimizer, lambda _: 1, last_epoch=last_epoch)
+    return LambdaLR(optimizer, get_constant_lambda, last_epoch=last_epoch)
```
Comment (sgugger) [inline src/transformers/optimization.py:64]:
```suggestion
    return LambdaLR(optimizer, _get_constant_lambda, last_epoch=last_epoch)
```

**Review by sgugger — COMMENTED** [review]
Thanks for opening a PR! Let's make the helper function private.

File: src/transformers/optimization.py (hunk 1 above)
Comment (Natyren) [reply src/transformers/optimization.py]: Done.

File: src/transformers/optimization.py (hunk 2 above)
Comment (Natyren) [reply src/transformers/optimization.py]: Delete it.

File: src/transformers/optimization.py (hunk 3 above)
Comment (Natyren) [reply src/transformers/optimization.py]: done refactoring based on function name

**Review by sgugger — APPROVED** [review]
Thanks!
